pyccel/ast/core.py

```diff
@@ -1632,17 +1555,18 @@ class ValuedArgument(Basic):
     """
 
     def __init__(self, expr, value, *, kwonly = False):
-        if isinstance(expr, str):
-            expr = Symbol(expr)
-
         # TODO should we turn back to Argument
 
-        if not isinstance(expr, (Symbol, Argument)):
+        if isinstance(expr, Argument):
+            self._name = expr.name
+        elif isinstance(expr, (str, PyccelSymbol)):
```

Comment:
`PyccelSymbol` is now a subclass of `str`

Reply:
0a67d23